sdk/storage/azure_storage_blob/src/models/blob_properties.rs

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+use azure_core::headers::Headers;
+
+/// Properties of an Azure Storage blob.
+///
+#[derive(Clone, Default, Debug)]
+pub struct BlobProperties {
+    pub access_tier_inferred: String,
+    pub access_tier: String,
+    pub blob_type: String,
+    pub content_length: String,
+    pub content_md5: String,
+    pub content_type: String,
+    pub creation_time: String,
+    pub etag: String,
+    pub last_modified: String,
+    pub lease_state: String,
+    pub lease_status: String,
+    pub server_encrypted: String,
+}
+
+impl BlobProperties {
+    pub(crate) fn build_from_response_headers(response_headers: &Headers) -> BlobProperties {
+        let mut properties = BlobProperties {
+            ..Default::default()
+        };
+        for (key, value) in response_headers.iter() {
+            match key.as_str() {
+                "content-length" => properties.content_length = String::from(value.as_str()),
```

Comment:
Some, like this, aren't strings. Why are you parsing them as such? You should use our header traits and even define a header. We also have a bunch of these already defined as `HeaderName`s that you should use for consistency.